curvefs/src/tools/query/curvefs_inode_s3infomap.cpp

```diff
@@ -0,0 +1,237 @@
+/*
+ *  Copyright (c) 2022 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: curve
+ * Created Date: 2022-03-25
+ * Author: chengyi01
+ */
+
+#include "curvefs/src/tools/query/curvefs_inode_s3infomap.h"
+
+#include <algorithm>
+#include <iostream>
+#include <memory>
+
+#include "src/common/string_util.h"
+
+DECLARE_string(metaserverAddr);
+DECLARE_string(poolId);
+DECLARE_string(copysetId);
+DECLARE_string(partitionId);
+DECLARE_string(fsId);
+DECLARE_string(inodeId);
+DECLARE_uint32(rpcStreamIdleTimeoutMs);
+
+namespace curvefs {
+namespace tools {
+namespace query {
+
+InodeBase TraslateInodeBase(
+    const curvefs::metaserver::GetOrModifyS3ChunkInfoRequest& source) {
+    InodeBase target;
+    target.set_poolid(source.poolid());
+    target.set_copysetid(source.copysetid());
+    target.set_partitionid(source.partitionid());
+    target.set_fsid(source.fsid());
+    target.set_inodeid(source.inodeid());
+    return target;
+}
+
+void InodeS3InfoMapTool::PrintHelp() {
+    CurvefsToolRpc::PrintHelp();
+    std::cout << " -poolId=" << FLAGS_poolId
+              << " -copysetId=" << FLAGS_copysetId
+              << " -partitionId=" << FLAGS_partitionId
+              << " -fsId=" << FLAGS_fsId << " -inodeId=" << FLAGS_inodeId
+              << " [-metaserverAddr=" << FLAGS_metaserverAddr
+              << " -rpcStreamIdleTimeoutMs=" << FLAGS_rpcStreamIdleTimeoutMs
+              << "]";
```

Comment:
Still need poolId copysetId and partitionId when query inode?

Reply:
> Still need poolId copysetId and partitionId when query inode?

no just fsid and copysetId